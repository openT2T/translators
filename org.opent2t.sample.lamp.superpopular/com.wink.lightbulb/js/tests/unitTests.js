var test = require('ava');
var translatorPath = require('path').join(__dirname, '..');
var runLampTests = require('opent2t-device-lamp/lampTests');
var deviceData = require('./devicedata');
var testData = require('./testdata');
var OpenT2TLogger = require('opent2t').Logger;
var logger = new OpenT2TLogger("info");
var MockHub = require('opent2t-device-winkhub/mockWinkHub');
var mockHub = new MockHub(logger, deviceData);

var settings = {
    logger,
    test,
    createTranslator: mockHub.createTranslator(translatorPath, deviceData.base_state.data.object_id, deviceData.base_state.data.uuid),
    setTestData: mockHub.setTestData,
    expectedExceptions : testData.expected_exceptions
};

// Run standard lamp unit tests
runLampTests(settings);