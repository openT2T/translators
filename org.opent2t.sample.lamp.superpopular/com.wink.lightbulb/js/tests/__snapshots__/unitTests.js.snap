exports[`GetPlatform 1`] = `"{\"opent2t\":{\"schema\":\"org.opent2t.sample.lamp.superpopular\",\"translator\":\"opent2t-translator-com-wink-lightbulb\",\"controlId\":\"F8CFB903-58BB-4753-97E0-72BD7DBC7933\",\"uuid\":\"F8CFB903-58BB-4753-97E0-72BD7DBC7933\"},\"availability\":\"online\",\"pi\":\"F8CFB903-58BB-4753-97E0-72BD7DBC7933\",\"mnmn\":\"Wink\",\"mnmo\":\"Light Bulb (Generic)\",\"n\":\"Mock Name\",\"rt\":[\"org.opent2t.sample.lamp.superpopular\"],\"entities\":[{\"n\":\"Mock Name\",\"rt\":[\"opent2t.d.light\"],\"di\":\"91ff489f-06e3-3ec5-6dcd-2649d14470f3\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/power\",\"rt\":[\"oic.r.switch.binary\"],\"if\":[\"oic.if.a\",\"oic.if.baseline\"]},{\"href\":\"/dim\",\"rt\":[\"oic.r.light.dimming\"],\"if\":[\"oic.if.a\",\"oic.if.baseline\"]},{\"href\":\"/colourChroma\",\"rt\":[\"oic.r.colour.chroma\"],\"if\":[\"oic.if.a\",\"oic.if.baseline\"]},{\"href\":\"/colourRGB\",\"rt\":[\"oic.r.colour.rgb\"],\"if\":[\"oic.if.a\",\"oic.if.baseline\"]},{\"href\":\"/colourMode\",\"rt\":[\"oic.r.mode\"],\"if\":[\"oic.if.a\",\"oic.if.baseline\"]}]}]}"`;

exports[`GetPlatformExpanded 1`] = `"{"opent2t":{"schema":"org.opent2t.sample.lamp.superpopular","translator":"opent2t-translator-com-wink-lightbulb","controlId":"F8CFB903-58BB-4753-97E0-72BD7DBC7933","uuid":"F8CFB903-58BB-4753-97E0-72BD7DBC7933"},"availability":"online","pi":"F8CFB903-58BB-4753-97E0-72BD7DBC7933","mnmn":"Wink","mnmo":"Light Bulb (Generic)","n":"Mock Name","rt":["org.opent2t.sample.lamp.superpopular"],"entities":[{"n":"Mock Name","rt":["opent2t.d.light"],"di":"91ff489f-06e3-3ec5-6dcd-2649d14470f3","icv":"core.1.1.0","dmv":"res.1.1.0","resources":[{"href":"/power","rt":["oic.r.switch.binary"],"if":["oic.if.a","oic.if.baseline"],"id":"power","value":false},{"href":"/dim","rt":["oic.r.light.dimming"],"if":["oic.if.a","oic.if.baseline"],"id":"dim","dimmingSetting":20,"range":[0,100]},{"href":"/colourChroma","rt":["oic.r.colour.chroma"],"if":["oic.if.a","oic.if.baseline"],"id":"colourChroma","hue":234,"saturation":80},{"href":"/colourRGB","rt":["oic.r.colour.rgb"],"if":["oic.if.a","oic.if.baseline"],"id":"colourRGB","rgbValue":[10,14,51],"range":[0,255]},{"href":"/colourMode","rt":["oic.r.mode"],"if":["oic.if.a","oic.if.baseline"],"modes":["hsb"],"supportedModes":["hsb","ct","rgb"],"id":"colourMode"}]}]}"`;
